Fails to build when building inside of jail using cmake

The file `unshield_config.h` isn't made available to source files when building in a build jail.

Steps to reproduce:

    git clone git://github.com/twogood/unshield.git
    cd unshield
    mkdir -p __build_jail__
    cd __build_jail__
    cmake ..
    make

Error message:

    Scanning dependencies of target convert_utf
    [  9%] Building C object lib/convert_utf/CMakeFiles/convert_utf.dir/ConvertUTF.c.o
    Linking C static library libconvert_utf.a
    [  9%] Built target convert_utf
    Scanning dependencies of target libunshield
    [ 18%] Building C object lib/CMakeFiles/libunshield.dir/bswap.c.o
    In file included from /tmp/git/unshield/lib/bswap.c:2:0:
    /tmp/git/unshield/lib/internal.h:6:29: fatal error: unshield_config.h: No such file or directory
     #include "unshield_config.h"
                                 ^
    compilation terminated.
    make[2]: *** [lib/CMakeFiles/libunshield.dir/bswap.c.o] Error 1
    make[1]: *** [lib/CMakeFiles/libunshield.dir/all] Error 2
    make: *** [all] Error 2

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,6 +58,7 @@ message(STATUS "BUILD_STATIC: ${BUILD_STATIC}")
 add_definitions(-DHAVE_CONFIG_H)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/lib/unshield_config.h.in ${CMAKE_CURRENT_BINARY_DIR}/lib/unshield_config.h)
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/libunshield.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libunshield.pc @ONLY)
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 # To force position independent code for static libs on Linux x64
 if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
@@ -68,4 +69,4 @@ add_subdirectory(lib)
 add_subdirectory(src)
 
 install(FILES man/unshield.1 DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/man/man1)
-install(FILES libunshield.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libunshield.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)

--- a/lib/internal.h
+++ b/lib/internal.h
@@ -3,7 +3,7 @@
 #define __internal_h__
 
 #include "libunshield.h"
-#include "unshield_config.h"
+#include "lib/unshield_config.h"
 
 #if HAVE_STDINT_H
 #include <stdint.h>

--- a/src/unshield.c
+++ b/src/unshield.c
@@ -13,7 +13,7 @@
 #include <unistd.h>
 #include "../lib/libunshield.h"
 #ifdef HAVE_CONFIG_H
-#include "../lib/unshield_config.h"
+#include "lib/unshield_config.h"
 #endif
 #if HAVE_FNMATCH
 #include <fnmatch.h>

